### PR TITLE
Resume activity indicator animations after page change

### PR DESCRIPTION
**Pull request by shichongrui**
affects: Parent
refs: MBL-13821
release note: Fixed bug where loading spinners would stop animating while loading

Test Plan:
 - Open a course with a syllabus
 - While the grades list is loading quickly swipe to
   the syllabus and then back to the grades list
 - The activity indicator should continue to animate
 - Pull to refresh
 - Quickly switch to the syllabus view and back
 - The refresh control should still be animating

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



#### Release Note: 
Fixed bug where loading spinners would stop animating while loading

#### Affected Apps: Parent

[MBL-13821](https://instructure.atlassian.net/browse/MBL-13821)

<details><summary>:x: XCTest failed: CoreTests/FileDetailsViewControllerTests/testSVG</summary>


    Assertion Failure: FileDetailsViewControllerTests.swift:214: Asynchronous wait failed: Exceeded timeout of 5 seconds, with unfulfilled expectations: "done".
    Assertion Failure: FileDetailsViewControllerTests.swift:216: XCTAssertFalse failed
    Assertion Failure: FileDetailsViewControllerTests.swift:217: XCTAssertTrue failed
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018955&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: Core/Core/Views/HorizontalMenuViewController.swift [hunk 1]
```diff
@@ -303,3 +310,7 @@ public extension HorizontalPagedMenuDelegate {
 
     var menuItemFont: UIFont { .scaledNamedFont(.semibold16) }
 }
+
+public protocol HorizontalPagedMenuItem {
+    func itemWillBeDisplayed()
```
Comment (bluwave) [inline Core/Core/Views/HorizontalMenuViewController.swift]: knitpick, how about `menuItemWillBeDisplayed` ?  Won't hold for this if you don't like the name.

File: Core/Core/Grades/GradesViewController.swift [hunk 2]
```diff
@@ -135,6 +136,17 @@ public class GradesViewController: UIViewController {
             view.setNeedsLayout()
         }
     }
+
```
Comment (bluwave) [inline Core/Core/Grades/GradesViewController.swift]: I find putting in an extension helps make it clearer that this comes from some protocol
```
extension GradesViewController: HorizontalPagedMenuItem {
    public func itemWillBeDisplayed() 
...
}
```

Again this is a knitpick.  Do it if you like

**Comment by bluwave** [comment]
Thanks for fixing this.

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1018958&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: Core/Core/Views/HorizontalMenuViewController.swift (hunk 1 above)
Comment (shichongrui) [reply Core/Core/Views/HorizontalMenuViewController.swift]: Yeah I like it. Will do.

File: Core/Core/Grades/GradesViewController.swift (hunk 2 above)
Comment (shichongrui) [reply Core/Core/Grades/GradesViewController.swift]: No problem with that. I'll do it.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1019118&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1019120&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|23
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**